src/main/io/osd.c

```diff
@@ -518,17 +518,22 @@ static bool osdDrawSingleElement(uint8_t item)
         }
 
     case OSD_CRAFT_NAME:
+        // This does not strictly support iterative updating if the craft name changes at run time. But since the craft name is not supposed to be changing this should not matter, and blanking the entire length of the craft name string on update will make it impossible to configure elements to be displayed on the right hand side of the craft name.
+        //TODO: When iterative updating is implemented, change this so the craft name is only printed once whenever the OSD 'flight' screen is entered.
+
         if (strlen(pilotConfig()->name) == 0) {
             strcpy(buff, "CRAFT_NAME");
         } else {
             for (unsigned int i = 0; i < MAX_NAME_LENGTH; i++) {
                 buff[i] = toupper((unsigned char)pilotConfig()->name[i]);
                 if (pilotConfig()->name[i] == 0) {
-                    buff[i] = SYM_BLANK;
+                    buff[i] = '\0';
```

Comment:
If `name[i]` is `0` and the `0` signifies the end of the string, then it should just break here without putting the `0` into `buff[i]`. Am I wrong?

Reply:
No you aren't. Didn't think that far.